quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliTlsCommandIT.java

```diff
@@ -0,0 +1,141 @@
+import static io.quarkus.ts.quarkus.cli.tls.surefire.TlsCommandTest.PASSWORD;
+import static io.quarkus.ts.quarkus.cli.tls.surefire.TlsCommandTest.TRUST_STORE_PATH;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.util.function.Function;
+
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import io.quarkus.logging.Log;
+import io.quarkus.test.bootstrap.QuarkusCliCommandResult;
+import io.quarkus.test.bootstrap.tls.GenerateCertOptions;
+import io.quarkus.test.bootstrap.tls.GenerateQuarkusCaOptions;
+import io.quarkus.test.bootstrap.tls.QuarkusTlsCommand;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.ts.quarkus.cli.tls.surefire.TlsCommandTest;
+
+@Tag("QUARKUS-4592")
+@Tag("quarkus-cli")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@QuarkusScenario
+@DisabledOnNative // Only for JVM verification
+public class QuarkusCliTlsCommandIT {
+
+    // locations are expected according to the Quarkus docs describes generation target
+    private static final File QUARKUS_BASE_DIR = new File(System.getenv("HOME"), ".quarkus");
+    private static final File DEV_CA_CERT_FILE = new File(QUARKUS_BASE_DIR, "quarkus-dev-root-ca.pem");
+    private static final File DEV_CA_PK_FILE = new File(QUARKUS_BASE_DIR, "quarkus-dev-root-key.pem");
+
+    @Inject
+    static QuarkusTlsCommand tlsCommand;
+
+    @Order(1)
+    @Test
+    public void generateQuarkusCa() {
+        // also prepares state for assertion in TlsCommandTest
+        deleteFileIfExists(DEV_CA_CERT_FILE);
+        deleteFileIfExists(DEV_CA_PK_FILE);
+        tlsCommand
+                .generateQuarkusCa()
+                .withOption(GenerateQuarkusCaOptions.TRUSTSTORE_LONG)
+                .withOption(GenerateQuarkusCaOptions.RENEW_SHORT)
+                .executeCommand()
+                .assertCommandOutputContains("Root CA certificate generated successfully")
+                .assertCommandOutputContains("Quarkus Development CA generated and installed")
+                .assertFileExistsStr(cmd -> cmd.getOutputLineRemainder("Truststore generated successfully:"));
+        assertTrue(DEV_CA_CERT_FILE.exists(),
+                "Quarkus CLI subcommand 'tls generate-quarkus-ca' didn't generate Quarkus DEV CA certificate");
+        assertTrue(DEV_CA_PK_FILE.exists(),
+                "Quarkus CLI subcommand 'tls generate-quarkus-ca' didn't generate Quarkus DEV CA private key");
+    }
+
+    @Order(2)
+    @Test
+    public void generateCertificate() {
+        // also prepares state for assertion in TlsCommandTest
+        var appSvcDir = tlsCommand.getApp().getServiceFolder().toAbsolutePath().toString();
```

Comment:
IMHO we don't need general var type here, I would prefer String, but it's not a blocker.
It's on more places in this PR, I'm not commenting on them, I'm just not fan of it.

Reply:
I will replace it, I don't mind. I just hope you won't read my other PRs because I use it as my goto type. Anyway, it is not important thing and I will address it. I don't see problem in changing it, it's just my habit.